src/test/addrman_tests.cpp

```diff
@@ -62,16 +95,16 @@ BOOST_AUTO_TEST_CASE(addrman_ports)
     // Set addrman addr placement to be deterministic.
     addrman.MakeDeterministic();
 
-    CNetAddr source = CNetAddr("252.2.2.2:8333");
```

Comment:
The following tests all resolve to true:

    BOOST_CHECK(CNetAddr("250.2.2.2:8333").IsIPv6());
    BOOST_CHECK(CNetAddr("250.2.2.2", 8333).IsIPv4());
    BOOST_CHECK(CService("250.2.2.2:8333").IsIPv4());
    BOOST_CHECK(CAddress(CService("250.2.2.2:8333")).IsIPv4());

This confusion arises because CNetAddr does not actually take a port number. I have a branch here that shows this behavior. 

https://github.com/EthanHeilman/bitcoin/commit/3f5c44080814b33257416b42a3ef941d726174fd

This behavior does not carry across into CService. Given these problems I need to write my commit. Thanks for your help @paveljanik  @MarcoFalke